File: 2541-minimum-operations-to-make-array-equal-ii/2541-minimum-operations-to-make-array-equal-ii.cpp
// If operations supported on both array then this will be perfect
/*
class Solution {
public:
    long long minOperations(vector<int>& nums1, 
                            vector<int>& nums2, 
                            int k, 
                            long long countItr = 0) {
        for (int i = 0; i < nums1.size(); i++) {
            auto diff = abs(nums1[i] - nums2[i]);
            if (diff % k != 0) return false;
            countItr += diff / k;
        }
        return countItr % 2 == 0 ? countItr / 2 : -1;
    }
};
*/

class Solution {
public:
    long long minOperations(vector<int>& nums1, vector<int>& nums2, int k,
                            long long absSum = 0, long long sum = 0) {        
        for (int i = 0; i < nums1.size(); i++) {
            auto diff = nums2[i] - nums1[i];
            if (k) {
                if (diff % k != 0) return -1;
                sum += diff;
                absSum += abs(diff) / k;
            } else if (diff) return -1;
        }
        return sum ? -1: absSum >> 1;
    }
};

File: leetcode/2541-minimum-operations-to-make-array-equal-ii/2541-minimum-operations-to-make-array-equal-ii.rs
impl Solution {
    pub fn min_operations(nums1: Vec<i32>, nums2: Vec<i32>, k: i32) -> i64 {
        let (mut steps, mut sum) = (0, 0);
        for i in 0..nums1.len() {
            let diff = nums2[i] - nums1[i];
            if k != 0 {
                if diff % k != 0 { return -1; }
                sum += diff;
                steps += (diff as i64).abs() / k as i64;
            } else if diff != 0 { return -1; }
        }
        if sum != 0 { return -1; }
        steps / 2
    }
}